src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -772,13 +772,13 @@ static int add_mounts(void) {
                 return btrfs_log_dev_root(LOG_ERR, r, "root file system");
         if (r < 0)
                 return log_error_errno(r, "Failed to determine block device of root file system: %m");
-        if (r == 0) { /* Not backed by block device */
+        if (r == 0 || r == 1) { /* Not backed by block device or the originating device couldn't be found */
```

Comment:
that's not right. get_block_device_harder() returns 1 in case it found block device info correctly. You are basically breaking the typical default case here...

Also, I'd prefer a simpler solution: let's simply change things so that /run/systemd/volatile-root takes precedence over the automatic logic. i.e. move the readlink_malloc() block below to the beginning of the function, and only search with get_block_device_harder() if that resulted in ENOENT